spark/src/test/scala/org/apache/comet/exec/CometJoinSuite.scala

```diff
@@ -192,6 +192,13 @@ class CometJoinSuite extends CometTestBase {
 
           // DataFusion HashJoin LeftAnti has bugs in handling nulls and is disabled for now.
           // left.join(right, left("_2") === right("_1"), "leftanti")
+
+          // Full join: struct key
```

Comment:
Thanks for adding the test. I wonder if we should make this more comprehensive to cover structs containing different types, nulls, and nested structs?

Also, what happens with structs containing unsupported types such as array and map? Do we still fall back for those? It would be good to have a test for this case as well.

Reply:
Good call will add more test cases. Created #797 to make it easier to create nulls of struct type.

> Also, what happens with structs containing unsupported types such as array and map?

Do you mean unsupported by comet here? I think the answer for `map` is that spark does not support joining on `map` so it will fail during the Spark planning stage. I will add a test for array by I guess that requires something like #793 first since we do not support reading arrays from parquet. Or is there some other way testing with arrays?

Reply:
Good point about map not being supported by Spark.

I think we should fall back for array for now because we don't really support array yet.

Reply:
I think we could improve our test framework to make it easier to test for fallback but it is possible with code like this (must be used after calling collect on a DataFrame). I think we can improve the tests in a future PR.

```
 val str =
          new ExtendedExplainInfo().generateExtendedInfo(df.queryExecution.executedPlan)
        assert(str.contains(expectedMessage))
```